src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexReplacement.cs

```diff
@@ -5,33 +5,37 @@
 // use when using regexs to search/replace, etc. It's logically
 // a sequence intermixed (1) constant strings and (2) group numbers.
 
-using System.Collections;
 using System.Collections.Generic;
 
 namespace System.Text.RegularExpressions
 {
     internal sealed class RegexReplacement
     {
-        /*
-         * Since RegexReplacement shares the same parser as Regex,
-         * the constructor takes a RegexNode which is a concatenation
-         * of constant strings and backreferences.
-         */
-        internal RegexReplacement(String rep, RegexNode concat, Dictionary<Int32, Int32> _caps)
-        {
-            StringBuilder sb;
-            List<String> strings;
-            List<Int32> rules;
-            int slot;
+        // Constants for special insertion patterns
+        internal const int Specials = 4;
+        internal const int LeftPortion = -1;
+        internal const int RightPortion = -2;
+        internal const int LastGroup = -3;
+        internal const int WholeString = -4;
 
-            _rep = rep;
+        private readonly String _rep;
+        private readonly List<String> _strings; // table of string constants
+        private readonly List<Int32> _rules;    // negative -> group #, positive -> string #
```

Comment:
These are all only used inside `RegexReplacement` (even in the desktop code) so they can be `private`.